src/Types.hs

```diff
@@ -59,7 +58,7 @@ makeClassy ''Env
 
 newtype App m a = App {
   unApp :: ReaderT Env m a
-} deriving (Functor, Applicative, Monad, MonadIO, MonadReader Env) -- all necessary for Katip
+} deriving newtype (Functor, Applicative, Monad, MonadIO, MonadReader Env) -- all necessary for Katip
```

Comment:
Sometimes it is nice to use `DerivingStrategies` to be clear that you're using `GeneralizedNewtypeDeriving`.